src/staticModelAdapt.cpp

```diff
@@ -0,0 +1,128 @@
+//
+// staticModelAdapt.cpp: A class containing parameters and functions to update
+// these parameters in the context of static Bayesian models. The methods to
+// estimate the empirical covariance and its Cholesky decomposition are applicable
+// for applications where the particle values are a vector of doubles. The methods
+// to compute the next 'temperature' are relevant to likelihood annealing SMC where the
+// power posteriors are defined by P_t(theta|y) \propto P(y|theta)^\gamma_t P(theta).
+// Here y is observed data, theta denotes the parameters of the model and gamma
+// denotes the 'temperatures' which start at 0 (the prior) and finish at 1
+// (the posterior).
+//
+// Copyright (C) 2017         Dirk Eddelbuettel, Adam Johansen and Leah South
+//
+// This file is part of RcppSMC.
+//
+// RcppSMC is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// RcppSMC is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with RcppSMC.  If not, see <http://www.gnu.org/licenses/>.
+
+#include "smctc.h"
+
+//! \file
+//! \brief This file contains the function definitions for adaptation
+//! in SMC for static Bayesian models.
+
+namespace smc {
+    /// Computes the difference between the conditional ESS given the specified temperature difference and the desired conditional ESS.
+    double staticModelAdapt::CESSdiff(const arma::vec & logweight, const arma::vec & loglike, double tempDiff, double desiredCESS){
+        double logsum1 = stableLogSumWeights(logweight + tempDiff*loglike);
+        double logsum2 = stableLogSumWeights(logweight + 2*tempDiff*loglike);
+
+        return exp(log(static_cast<double>(logweight.n_rows)) + 2*logsum1 - logsum2) - desiredCESS;
+    }
+    
+    ///Performs the bisection method to find the temperature within (temp_curr,1) which gives the desired conditional ESS.
+    double staticModelAdapt::bisection(double curr, const arma::vec & logweight, const arma::vec & loglike, const double & desiredCESS){
+        double a = curr;
+        double b = 1;
+        double f_a = CESSdiff(logweight,loglike,a-curr,desiredCESS);
+        double f_b = CESSdiff(logweight,loglike,b-curr,desiredCESS);
+        if (f_a*f_b>0){
+            Rcpp::stop("Bisection method to choose the next temperature failed");
+        } else{
+            double m, f_m, err;
+            m = (a+b)/2.0;
+            f_m = CESSdiff(logweight,loglike,m-curr,desiredCESS);
+            err = 10;
+            while (err > 0.01){
```

Comment:
Thanks Adam for this and for all of your other good points. I should have picked up on at least some of these myself -- sorry about that. I've changed this and a couple of other minor things (like writing 1.0 instead of 1 if it's a double) and will do another commit when the check finishes.